docs/walkthrough.md

```diff
@@ -0,0 +1,500 @@
+  # pod and namespace information to each collected sample, so
+  # that they'll be exposed in the custom metrics API automatically.
+  - source_labels: [__meta_kubernetes_namespace]
+    action: replace
+    target_label: namespace
+  - source_labels: [__meta_kubernetes_pod_name]
+    action: replace
+    target_label: pod
+  # these labels tell Prometheus to look for
+  # prometheus.io/{scrape,path,port} annotations to configure
+  # how to scrape
+  - source_labels: [__meta_kubernetes_pod_annotation_prometheus_io_scrape]
+    action: keep
+    regex: true
+  - source_labels: [__meta_kubernetes_pod_annotation_prometheus_io_path]
+    action: replace
+    target_label: __metrics_path__
+    regex: (.+)
+  - source_labels: [__address__, __meta_kubernetes_pod_annotation_prometheus_io_port]
+    action: replace
+    regex: ([^:]+)(?::\d+)?;(\d+)
+    replacement: $1:$2
+    target_label: __address__
+  - source_labels: [__meta_kubernetes_pod_annotation_prometheus_io_scheme]
+    action: replace
+    target_label: __scheme__
+    regex: (.+)
+```
+
+</details>
+
+Place your full configuration (it might just be the above) in a file (call
+it `prom-cfg.yaml`, for example), and then create a ConfigMap containing
+it:
+
+```shell
+$ kubectl create configmap prometheus --from-file=prometheus.yml=prom-cfg.yaml
+```
+
+You'll be using this later when you launch Prometheus.
+
+### Launching the Deployment ###
+
+It's generally easiest to launch the adapter and Prometheus as two
+containers in the same pod.  You can use a deployment to manage your
+adapter and Prometheus instance.
+
+Start out with a fairly straightforward Prometheus deployment using the
+ConfigMap from above, and proceed from there:
+
+<details>
+
+<summary>prom-adapter.deployment.yaml [Prometheus only]</summary>
+
+```yaml
+apiVersion: apps/v1beta1
+kind: Deployment
+metadata:
+  name: prometheus
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: prometheus
+  template:
+    metadata:
+      labels:
+        app: prometheus
+    spec:
+      containers:
+      - image: prom/prometheus:v1.6.1
+        name: prometheus
+        args:
+        - -storage.local.retention=6h
+        - -storage.local.memory-chunks=500000
+        # point prometheus at the configuration that you mount in below
+        - -config.file=/etc/prometheus/prometheus.yml
+        ports:
+        # this port exposes the dashboard and the HTTP API
+        - containerPort: 9090
+          name: prom-web
+          protocol: TCP
+        volumeMounts:
+        # you'll mount your ConfigMap volume at /etc/prometheus
+        - mountPath: /etc/prometheus
+          name: prom-config
+      volumes:
+      # make your configmap available as a volume, so that you can
+      # mount in the Prometheus config from earlier
+      - name: config-volume
+        configMap:
+          name: prometheus
+```
+
+</details>
+
+Save this file (for example, as `prom-adapter.deployment.yaml`).  Now,
```

Comment:
I wouldn't bundle prometheus and the adapter in the same pod in this walkthrough.
Deploy those separately

Here, the "Deploy the Custom Metrics Adapter for Prometheus" topic should start